File: Python/thread1.py
from threading import Thread
from time import sleep
from threading import Lock

class my_thread(Thread):

	def __init__(self,name,msg):
		Thread.__init__(self)
		self.name = name
		self.msg = msg

	def run(self):
		l.acquire()
		i = 0
		while i<len(self.msg):
			k = open("thread.txt",'a')
			k.write(self.msg[i])
			print self.msg[i]
			i = i+1
			sleep(1)
			k.close()
		l.release()		

l = Lock()

t1 = my_thread('thread_1','Valar Morghulis')
t1.start()	#main thread runs internally which is the benifit of using threading module but only one thread runs. Use t1.start()
#t1.join()

t2 = my_thread('thread_2','A lannister always pays his debts')
t2.start()
#t2.join()

File: Python/thread1.c
#include "pthread.h"
#include "sys/types.h"

struct xxx
{
	char name[20];
	int ctime;
};

bbsr(struct xxx*p)
{
	int i;
	for(i = 0;i < p->ctime;i++)
	{
		printf("I am %s\n",p->name);
		sleep(1);
	}
}

main()
{
	struct xxx a = {
			.name = "thread_1",
			.ctime = 5
			};
	struct xxx b = {
			.name = "thread_2",
			.ctime = 10
			};
	struct xxx c = {
			.name = "thread_3",
			.ctime = 15
			};

	pthread_t t1,t2,t3;
	
	pthread_create(&t1,0,bbsr,&a);
	
	pthread_create(&t2,0,bbsr,&b);
	pthread_create(&t3,0,bbsr,&c);

	while(1)
	{
		printf("I am MAIN Thread!!\n");
		sleep(1);
	}
}
